### PR TITLE
Id reassign

### PR DESCRIPTION
**Pull request by AdaRoseCannon**
Needs tests writing.

### CODE REVIEW

**Comment by JakeChampion** [comment]
Assigned @seanmtracey as he implemented this logic in Screens repo

**Comment by seanmtracey** [comment]
Just thinking, shouldn't the tests for collisions reassignment be server-side? After all, that's where it's ultimately decided that a screen should be reassigned an ID and that ID is generated.

**Comment by AdaRoseCannon** [comment]
Ah but the client side part of the api needs to be tested.
It needs to receive the api request and then update it's id.

**Comment by AdaRoseCannon** [comment]
Fixes #9

File: build/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ftlabs-screens-viewer",
-  "version": "1.0.10",
+  "version": "1.0.13",
```
Comment (JakeChampion) [inline build/package.json:3]: 3 minor versions?

File: lib/index.js [hunk 2]
```diff
@@ -33,6 +33,24 @@ function Viewer(host) {
 
 	socket.on('connect', () => this.connectionState = true );
 
+	socket.on('reassign', data => {
+
+		console.log("Reassign", data);
```
Comment (JakeChampion) [inline lib/index.js:38]: Should we log in production?

File: lib/index.js [hunk 3]
```diff
@@ -114,8 +132,27 @@ util.inherits(Viewer, EventEmitter);
 module.exports = Viewer;
 
 Viewer.prototype.update = function update(newdata) {
+
+	const olddata = this.getData();
```
Comment (JakeChampion) [inline lib/index.js]: Could we use camelCase for the variable name

File: lib/index.js [hunk 4]
```diff
@@ -114,8 +132,27 @@ util.inherits(Viewer, EventEmitter);
...
+
+	if(!olddata.idUpdated && !newdata.idUpdated){
+		newdata.idUpdated = Date.now();
```
Comment (JakeChampion) [inline lib/index.js]: Same here

**Comment by JakeChampion** [comment]
Tiny comments, other than those, LGTM :+1:

File: lib/index.js (hunk 2 above)
Comment (AdaRoseCannon) [inline lib/index.js:38]: This was from the original implementation. I assume there was a reason it was there. Also we already do log lots in production. Seems harmless to me.

File: build/package.json (hunk 1 above)
Comment (AdaRoseCannon) [inline build/package.json:3]: Yeah there were a few dead end deploys with now fixed bugs. I deleted the tags but don't like rolling back version numbers.

File: lib/index.js (hunk 3 above)
Comment (AdaRoseCannon) [inline lib/index.js]: :+1: Should have spotted that when porting it in, nice spot.

File: build/package.json (hunk 1 above)
Comment (JakeChampion) [inline build/package.json:3]: Is it still rolling back if the versions were never released?

Comment (AdaRoseCannon) [inline build/package.json:3]: They were released and I tested them and found bugs. So I deleted the tag to stop them being available.

**Pull request merged**